crates/volta-core/src/hook/tool.rs

```diff
@@ -43,56 +46,73 @@ impl DistroHook {
 pub enum MetadataHook {
     Prefix(String),
     Template(String),
-    Bin(String),
+    Bin { bin: String, base_path: PathBuf },
 }
 
 impl MetadataHook {
     /// Performs resolution of the metadata URL based on the given default file name
     pub fn resolve(&self, filename: &str) -> Fallible<String> {
-        match self {
-            &MetadataHook::Prefix(ref prefix) => Ok(format!("{}{}", prefix, filename)),
-            &MetadataHook::Template(ref template) => Ok(template
+        match &self {
+            MetadataHook::Prefix(prefix) => Ok(format!("{}{}", prefix, filename)),
+            MetadataHook::Template(template) => Ok(template
                 .replace(ARCH_TEMPLATE, ARCH)
                 .replace(OS_TEMPLATE, OS)),
-            &MetadataHook::Bin(ref bin) => execute_binary(bin, None),
+            MetadataHook::Bin { bin, base_path } => execute_binary(bin, base_path, None),
         }
     }
 }
 
 /// Execute a shell command and return the trimmed stdout from that command
-fn execute_binary(bin: &str, extra_arg: Option<String>) -> Fallible<String> {
+fn execute_binary(bin: &str, base_path: &Path, extra_arg: Option<String>) -> Fallible<String> {
     let mut trimmed = bin.trim().to_string();
     let mut words = trimmed.parse_cmdline_words();
-    let cmd = if let Some(word) = words.next() {
-        word
-    } else {
-        throw!(ErrorDetails::InvalidHookCommand {
-            command: String::from(bin.trim()),
-        })
+    let cmd = match words.next() {
+        Some(word) => {
+            // Treat any path that starts with a '.' as a relative path
+            if word.starts_with('.') {
```

Comment:
I'm concerned about this implementation for the reason that, though it's not advised, you *could* have an item somewhere in your `PATH` which is *named* `.foo` and that's what you're trying to run. Should we make this check be against (the OS-equivalent of) `"./"` instead of `'.'`?

```rust
lazy_static! {
    static ref REL_PATH: String = format!(".{}", std::path::MAIN_SEPARATOR);
}

if word.starts_with(REL_PATH) {
  // ...
}
```

Reply:
Hmm, that's true, but we also need to account for a path that starts with `../` (or the OS-equivalent). We can pretty easily check both, but now I'm curious if there are any other corner cases we need to consider.

Reply:
Ah, yep, good point.

Reply:
I couldn't think of any other edge cases, so I think we're okay just checking for the two types of relative path: `./` and `../`, normalized to the OS path separator.

Reply:
Very good!